### PR TITLE
Doc hyperlink

### PR DESCRIPTION
**Pull request by willghatch**
## Summary

This adds cross reference hyperlinking to the documentation.

## Some questions:

* It's been... a while since I've done anything serious in Javascript.  What is our preferred method of returning maybe something?  In this I returned an `undefined`, but I am doubtful that it's the best option.  Maybe `false` would be good, but `false == 0` and similar punning makes me less confident using false in JS than I would in scheme.  Any thoughts?
* I didn't see anything in the onboarding docs about how I should go about submitting PRs.  Is this good?

### CODE REVIEW

File: docs/dev/generator.js [hunk 1]
```diff
@@ -536,6 +541,17 @@ const processMd = async ({baseConfig, relDir, in_folder, iPath, oPath}) => {
   theader.remove();
 
   // Process code snippets.
+  const linkifySpan = (s, language) => {
+    const text = s.textContent
+    const ref = xrefGetMaybe(language == "reach" ? "rsh" : language, text);
```
Comment (jeapostrophe) [inline docs/dev/generator.js:546]: You should make a normalization dictionary and error if there's no normalization and/or consult the `aliases` field of Shiki's bundled languages (see the `sher` defn)

File: docs/dev/generator.js [hunk 2]
```diff
@@ -653,6 +672,9 @@ const processMd = async ({baseConfig, relDir, in_folder, iPath, oPath}) => {
     const s = doc.createElement('span');
     s.classList.add('snip');
     s.appendChild(mkEl(hc));
+    for (const is of s.querySelectorAll('span')) {
```
Comment (jeapostrophe) [inline docs/dev/generator.js:675]: Shame! No copying code!

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
`undefined` is great.

Does it look nice? Can you include a screenshot? I'm surprised you don't need to change the CSS to not get underlines and boldness

**Comment by jeapostrophe** [comment]
This PR is fine. One thing though is that the CI will run if you commit to a branch on `reach-sh:master` rather than your own fork

**Comment by willghatch** [comment]
So in that screenshot, the upper code snippet in Reach has clickable `const`, `export`, `App`, `Participant`, etc, and the `interact` inline code snippet between the two blocks is clickable.  However, the lower code block is in Javascript, and none of the keywords have defined refs, so nothing in that block is clickable.

**Comment by jeapostrophe** [comment]
That's cool, I like it.

File: docs/dev/generator.js [hunk 3]
```diff
@@ -536,6 +552,22 @@ const processMd = async ({baseConfig, relDir, in_folder, iPath, oPath}) => {
   theader.remove();
 
   // Process code snippets.
+  const linkifySpans = (elem, language) => {
+    const linkifySpan = (s, language) => {
+      const text = s.textContent
+      const ref = xrefGetMaybe(language == "reach" ? "rsh" : language, text);
```
Comment (jeapostrophe) [inline docs/dev/generator.js:558]: I don't think this `?:` is needed because of the normalization

File: docs/dev/generator.js [hunk 4]
```diff
@@ -53,21 +53,51 @@ const fail = (...args) => {
...
+  },
+];
+const languageNormalizeDict = {};
+for (const l of languages) {
+  languageNormalizeDict[l.id] = l.id;
+  for (const alias of (l.aliases || [])) {
+    languageNormalizeDict[alias] = l.id;
+  }
+}
+const normalizeScope = (s) => languageNormalizeDict[s] || s;
```
Comment (jeapostrophe) [inline docs/dev/generator.js:79]: I think you should error if there's no normalization and we'd fix it

**Comment by willghatch** [comment]
After figuring out `scrot`'s options, here is a screenshot showing my cursor hovering over a link:

doc-hyperlink-screenshot-2

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
Does any JS get linked? What about go and python? (See the RPC pages)

File: docs/dev/generator.js (hunk 3 above)
Comment (willghatch) [reply docs/dev/generator.js:558]: Oh, yeah.

File: docs/dev/generator.js (hunk 4 above)
Comment (willghatch) [reply docs/dev/generator.js:79]: So one thing that that allows through is that a bunch of things are added with a literal 'h' scope.  Also with literal 'term' scope.  These are not languages that Shiki knows about.  So this cross reference stuff is used for both code cross references and document cross references.

**Comment by willghatch** [comment]
> Does any JS get linked? What about go and python? (See the RPC pages)

Yes, though there aren't many JS, go, or python references defined.  The RPC pages each have a couple links.

File: docs/dev/generator.js (hunk 4 above)
Comment (willghatch) [reply docs/dev/generator.js:79]: Shall we add a list of literals like 'h' and 'term' to the normalizer, and error if it's not a language name in the shiki list or a known document scope literal?

Comment (jeapostrophe) [reply docs/dev/generator.js:79]: I see. Yea, I think you should add them.

**Comment by willghatch** [comment]
Ok, it now fails for scopes that can't be normalized (by being document scopes or in the language list).

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**